bots/tester/src/test/java/org/openjdk/skara/bots/tester/InMemoryHostedRepository.java

```diff
@@ -172,6 +172,14 @@ public List<HostedBranch> branches() {
         return List.of();
     }
 
+    @Override
+    public void protectBranchPattern(String ref) {
+    }
```

Comment:
The name of this method is protectBranchPattern, but I think it actually protects a specific branch rather than a pattern.

Reply:
Hm, on GitLab the underlying call takes a pattern, but now that I think about it, on GitHub it doesn't. There you have to use GraphQL to add pattern based protection. I should change this method to reflect that. We aren't using patterns anyway.

Reply:
I take that back. I see now that I chose not to implement this on GitHub at all since we don't need it. This does take a branch pattern, which are supported by both GitHub and GitLab in the web UI.

Reply:
> I take that back. I see now that I chose not to implement this on GitHub at all since we don't need it. This does take a branch pattern, which are supported by both GitHub and GitLab in the web UI.

Got it! It makes sense